src/main/java/com/example/WMS/services/servicesimpl/IncomingGoodsServicesImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.example.WMS.services.servicesimpl;
+
+import com.example.WMS.dtos.*;
+import com.example.WMS.entity.IncomingGoods;
+import com.example.WMS.entity.ProductDetails;
+import com.example.WMS.exceptions.OrderNotFound;
+import com.example.WMS.exceptions.ProductNotPresent;
+import com.example.WMS.handlers.ProductDetailsHandler;
+import com.example.WMS.handlers.IncomingGoodsHandler;
+import com.example.WMS.services.IncomingGoodsServices;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+@Slf4j
+public class IncomingGoodsServicesImpl implements IncomingGoodsServices {
+
+    @Autowired
+    private IncomingGoodsHandler incomingGoodsHandler;
+
+    @Autowired
+    private ProductDetailsHandler productDetailsHandler;
+
+    public ValidationDto createIncomingGoods(AddStockDto addStockDto) {
+        String uniqueID = UUID.randomUUID().toString();
```

Comment:
What is the purpose of this Unique ID ??
Rather expect ID from customer 
and also maintain the idempotent check with that ID

Reply:
what would be this unique ID for incoming goods? I have made a unique ID for incoming goods and returned the value on successful storage of incoming goods. For adding products i have used the product name as the ID and validated that the product name is always unique while inserting new products into the CSV file.